hub/saml-engine/src/main/java/uk/gov/ida/hub/samlengine/validation/country/EidasAttributeStatementAssertionValidator.java

```diff
@@ -106,4 +110,21 @@ private void validateAttributes(Assertion assertion) {
             }
         }
     }
+
+    private QName getAttributeValueSchemaType(Response response, Attribute attribute) {
+        final QName attributeValueSchemaType = attribute.getAttributeValues().get(0).getSchemaType();
+        if (attributeValueSchemaType.getNamespaceURI().isEmpty()) {
+            Optional<Namespace> matchingPrefix = response.getNamespaces()
+                    .stream()
+                    .filter(ns -> Objects.equals(ns.getNamespacePrefix(), attributeValueSchemaType.getPrefix()))
+                    .findFirst();
+            if (matchingPrefix.isPresent()) {
```

Comment:
` .findFirst()` returns Optional so you can chain `ifPresent` directly on the result - no need for the second `if` statement

Reply:
I've now used Optional.map which is similar to ifPresent but allows a return